lib/samples/create_mobile_geodatabase/create_mobile_geodatabase_sample.dart

```diff
@@ -0,0 +1,353 @@
+//
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'dart:io';
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:flutter/material.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:share_plus/share_plus.dart';
+
+import '../../utils/sample_state_support.dart';
+
+class CreateMobileGeodatabaseSample extends StatefulWidget {
+  const CreateMobileGeodatabaseSample({super.key});
+
+  @override
+  State<CreateMobileGeodatabaseSample> createState() =>
+      _CreateMobileGeodatabaseSampleState();
+}
+
+class _CreateMobileGeodatabaseSampleState
+    extends State<CreateMobileGeodatabaseSample> with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // Declare a map to be loaded later.
+  late final ArcGISMap _map;
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // A mobile Geodatabase to be created and shared.
+  Geodatabase? _geodatabase;
+  // A feature table to store the location history.
+  GeodatabaseFeatureTable? _featureTable;
```

Comment:
I think it would make more sense for this to be a `late final GeodatabaseFeatureTable _featureTable;`. That would follow the pattern for `_map`. It should be protected by the `_ready` mechanism.